blocks/v2-article-search/v2-article-search.js

```diff
@@ -0,0 +1,273 @@
+const searchInputExpanded = `${blockName}__search-input--expanded`;
+
+await getPlaceholders();
+const filterDefault = getTextLabel('filterDefault');
+const searchPlaceholder = getTextLabel('searchPlaceholder');
+
+const locale = getLocale();
+const language = locale.split('-')[0].toUpperCase();
+
+const currentURL = new URL(window.location.href);
+const magazinePath = '/news-and-stories/volvo-trucks-magazine/';
+const magazineParam = '?search=&category=&topic=&truck=';
+currentURL.pathname = magazinePath;
+currentURL.search = magazineParam;
+
+const getTopics = async (props = {}) => {
+  const { limit = 1, offset = 0, article = {} } = props;
+  const variables = {
+    tenant: TENANT,
+    language,
+    limit,
+    offset,
+    category: 'magazine',
+    article,
+    sort: 'BEST_MATCH',
+    facets: [
+      'ARTICLE',
+      'TOPIC',
+      'TRUCK',
+    ],
+  };
+
+  try {
+    const rawData = await fetchData({
+      query: topicSearchQuery(),
+      variables,
+    });
+
+    const querySuccess = rawData && rawData.data && rawData.data.edsrecommend;
+
+    if (!querySuccess) {
+      return [];
+    }
+
+    const [trucks, topics, articles] = querySuccess.facets;
+    return {
+      truck: [...new Set(trucks.items.map((item) => item.value.trim()))],
+      topic: [...new Set(topics.items.map((item) => item.value.trim()))],
+      category: [...new Set(articles.items.map((item) => item.value.trim()))],
+    };
+  } catch (error) {
+    console.error('Error fetching data:', error);
+    throw error;
+  }
+};
+
+const buildFilterElement = () => document.createRange().createContextualFragment(`
+  <div class="${blockName}__filter-container">
+    <form class="${blockName}__filter">
+      <fieldset class="${blockName}__fieldset">
+        <div class="${blockName}__filter-dropdown">
+          ${filterDefault} <span class="${blockName}__filter-dropdown-icon">
+            <span class="icon icon-chevron-down"></span>
+          </span>
+        </div>
+        <div class="${blockName}__filter-input">
+          <span class="icon icon-search-icon"></span>
+          <input type="text" id="search" name="search" placeholder="${searchPlaceholder}" />
+          <label for="search" class="${blockName}__input-label">${searchPlaceholder}</label>
+          <button type="button" class="${blockName}__clear-button">clear</button>
+          <span class="icon icon-close"></span>
+        </div>
+      </fieldset>
+    </form>
+    <div class="${blockName}__filter-list-wrapper"></div>
+  </div>
+`);
+
+const buildBulletList = (allTopics) => Object.keys(allTopics).map((key) => {
+  const items = allTopics[key];
+  return items.map((item) => {
+    const param = item.toLowerCase().replace(/\s/g, '-');
+    currentURL.search = magazineParam;
+    currentURL.searchParams.set(key, param);
+    return `<li class="${blockName}__filter-item">
+      <a href="${currentURL.href.replace('#', '')}" target="_blank"
```

Comment:
should this always open in a new tab? 
I see this AC: 
_Content authors should be able to determine whether selecting a new filter tag will open the category content in a new browser tab or keep the user on the current page._

Also, when going back to the original page, the topic remains selected in black

Reply:
removed target blank